### PR TITLE
feat: add support for Ponder ESLint config

### LINKED ISSUE #305: Warning on missing promise await on upsert operations
It's not uncommon for new users of Ponder (including myself 😅) to overlook the need for awaiting entities' upsert operations. This oversight can lead to issues where the node process doesn't wait for the new record to be created before proceeding with the syncing process. As a result, the subsequent sync cycle may attempt to [update an instance from the previous sync cycle](https://github.com/0xOlias/ponder/blob/main/packages/core/src/user-store/postgres/store.ts#L324).

Example:
```typescript
ponder.on("Kongz:Transfer", async ({ event, context }) => {
  context.entities.Account.upsert({
    // ^? this is missing an await
    id: event.params.from,
    create: {},
    update: {},
  });
});
```

Generates this error:
```
Error: Cannot update an instance in the past
```

To address this issue, one potential solution is to introduce a lint rule that can alert users about the absence await keyword misisng. Similar to React's `exhaustive-deps` warning.

### LINKED ISSUE #315: Create shared ESLint configuration for Ponder
It'd be great if we can have a base ESLint config that users can extend from. This file will have all the configs we believe are useful for Ponder apps.

### PR DESCRIPTION
**Pull request by eliobricenov**
closes #315

### CODE REVIEW

File: packages/create-ponder/src/index.ts [hunk 1]
```diff
@@ -201,6 +202,17 @@ export const run = async (
     prettier.format(tsConfig, { parser: "json" })
   );
 
+  const eslintConfig = `
+    {
+      "extends": "eslint-config-ponder"
```
Comment (0xOlias) [inline packages/create-ponder/src/index.ts:207]: Pretty sure this can just be `"extends": "ponder"` 🤓

**Review by 0xOlias — COMMENTED** [review]
Just pushed a commit with some small tweaks to get it working in the monorepo

File: packages/create-ponder/src/index.ts (hunk 1 above)
Comment (eliobricenov) [reply packages/create-ponder/src/index.ts:207]: yes!

File: examples/ethfs/package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   },
   "devDependencies": {
     "abitype": "^0.8.11",
+    "eslint-config-ponder": "workspace:latest",
```
Comment (0xOlias) [inline examples/ethfs/package.json:14]: Should also add `eslint` as a dev dependency for all the examples

File: examples/token-erc20/package.json [hunk 3]
```diff
@@ -4,13 +4,16 @@
   "scripts": {
     "dev": "ponder dev",
     "start": "ponder start",
-    "codegen": "ponder codegen"
+    "codegen": "ponder codegen",
+    "lint": "eslint . --debug"
```
Comment (0xOlias) [inline examples/token-erc20/package.json:8]: Can remove the `--debug`. Also, all of the examples should have this lint script as I mentioned below.

File: packages/create-ponder/src/index.ts [hunk 4]
```diff
@@ -159,12 +165,14 @@ export const run = async (
       "scripts": {
         "dev": "ponder dev",
         "start": "ponder start",
+        ${options.eslint ? `"lint": "eslint .",` : ""}
```
Comment (0xOlias) [inline packages/create-ponder/src/index.ts:168]: Oh this is great! I'd add this lint script to all of the examples/ projects too. In general I think it's nice for the examples/ projects to look like they were just generated using create-ponder.

File: packages/eslint-config-ponder/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "eslint-config-ponder",
+  "version": "0.0.81",
```
Comment (0xOlias) [inline packages/eslint-config-ponder/package.json:3]: Should make this equal to whatever the latest version is on main (for @ponder/core and create-ponder).

File: packages/eslint-config-ponder/package.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "eslint-config-ponder",
```
Comment (0xOlias) [inline packages/eslint-config-ponder/package.json:2]: I think we want to call the package `eslint-config-ponder` as shown here. I saw in a few other places it's referenced as `@ponder/eslint-config`, so I think you'll have to change those.

**Review by 0xOlias — COMMENTED** [review]
Some nits

File: packages/eslint-config-ponder/package.json (hunk 6 above)
Comment (eliobricenov) [reply packages/eslint-config-ponder/package.json:2]: jfc - that's what I get for coding before the first cup of coffee

**Review by 0xOlias — APPROVED** [review]

**Pull request merged**